Update filter.md to reflect new usage and improve wording (#81)
retest/docs: docs/recheck/usage/filter.md

### What Are Filters?

Filters can be used for different purposes, most notably to reduce the noise mentioned above within a report. They can accept two different arguments:

> 1. **Element**: Filter applies to _all attributes_ of an element and all of its child elements. The element in question can be identified by a wide range of identifying attributes, like XPath, HTML-id, retestId, CSS-class, HTML-name, etc.

Reviewer: > like XPath, HTML-id, retestId, CSS-class, HTML-name, etc

This is described below. Thus I would keep this here generic...

```suggestion
1. **Element**: Filter applies to _all attributes_ of an element and all of its child elements. The element in question can be identified by a wide range of identifying attributes, which are specified by the [extension](../introduction/installation) .
```

> 1. **Element**: Filter applies to _all attributes_ of an element and all of its child elements. The element in question can be identified by a wide range of identifying attributes, like XPath, HTML-id, retestId, CSS-class, HTML-name, etc.

Reviewer: Come to think of it: If we specify below the attribute, we might as well just describe the element as single element. The Matching via `matcher:` specifies how the elements are searched... Thus the concrete element would only be `id=foo`, `type=bar` which is the inner part of the matcher.

Thus I would move this part for specifying the child elements below with the element matching.
